In [18]:
# Dependencies and Setup
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import requests
import gmaps
import os
import json

# Import API key
from config import g_key
gmaps.configure(api_key=g_key)

In [25]:
#Store Part I results (from WeatherPy) into DataFrame and Load the csv exported in Part I to a DataFrame.

weather_csv_file = "../WeatherPy/Output/weather_df.csv"

weather_df = pd.read_csv(weather_csv_file)

weather_df.head()

Unnamed: 0  City name Country  Cloudiness       Dates  Humidities     Lats  \
0           0  dhidhdhoo      MV          95  1631668720          76   6.8833   
1           1   sterling      US          20  1631668596          82  39.0062   
2           2       asau      RO           7  1631668613          97  46.4333   
3           3    qaanaaq      GL          85  1631668721          91  77.4840   
4           4    mataura      NZ          10  1631668721          49 -46.1927   

       Lngs  Max_temps  Wind_speeds  
0   73.1000      83.23        15.37  
1  -77.4286      80.89         4.61  
2   26.4000      56.32         3.11  
3  -69.3632      17.69         5.82  
4  168.8643      49.48         5.14


Out[2]:
City	Cloudiness	Country	Date	Humidity	Latitude	Longitude	Max Temp	Wind Speed
0	hasaki	75.0	JP	1.580018e+09	86.0	35.73	140.83	48.00	5.82
1	oktyabrskoye	0.0	RU	1.580018e+09	85.0	43.06	44.74	19.00	6.71
2	atuona	37.0	PF	1.580018e+09	76.0	-9.80	-139.03	81.00	14.47
3	rikitea	24.0	PF	1.580018e+09	78.0	-23.12	-134.97	78.73	11.10
4	bredasdorp	69.0	ZA	1.580018e+09	77.0	-34.53	20.04	63.00	12.75

In [26]:

# Humidity Heatmap
# Configure gmaps.
# Use the Lat and Lng as locations and Humidity as the weight.

# Configure gmaps with API key.
gmaps.configure(api_key=g_key)

# Also, handle NaN values
weather_df = weather_df.dropna()

# Convert Humidity to float and store
humidity = weather_df["Humidities"].astype(float)
humidity

# Store 'Latitude' and 'Longitude' into  locations. 
locations = weather_df[["Lats", "Lngs"]].astype(float)
locations
# Create a humidity Heatmap layer
figure_layout = {
    'width': '1000px',
    'height': '800px',
    'border': '1px solid black',
    'padding': '1px'
}
fig = gmaps.figure(layout=figure_layout)
fig
# Add Heatmap layer to map.
heat_layer = gmaps.heatmap_layer(locations, weights=humidity, 
                                 dissipating=False, max_intensity=max(weather_df["Humidities"]),
                                 point_radius = 2)

fig.add_layer(heat_layer)

fig



Figure(layout=FigureLayout(border='1px solid black', height='800px', padding='1px', width='1000px'))

In [ ]:
# Narrow down the DataFrame to find your ideal weather condition.
filtered_weather_df = weather_df

# A max temperature lower than 80 degrees but higher than 70.
filtered_weather_df = filtered_weather_df.loc[(filtered_weather_df["Max_temps"] < 80) & (filtered_weather_df["Max_temps"] > 70)]

# Wind speed less than 10 mph.
filtered_weather_df = filtered_weather_df.loc[filtered_weather_df["Wind_speeds"] < 10]

# Zero cloudiness.
filtered_weather_df = filtered_weather_df.loc[filtered_weather_df["Cloudiness"] == 0]

# Drop any rows with null values
filtered_weather_df = filtered_weather_df.dropna()

filtered_weather_df

Unnamed: 0                  City name Country  Cloudiness       Dates  \
24           24                     bismil      TR           0  1594849774   
96           96                 rio muerto      GR           0  1594849303   
158         158                    aljezur      PT           0  1594849245   
181         181          estacion coahuila      PH           0  1594849310   
201         201                      pilar      BR           0  1594849790   
276         276                      salym      RU           0  1594849799   
313         313                 rio grande      GR           0  1594849802   
396         396  santa isabel do rio negro      PH           0  1594849810   
435         435                     artyom      AZ           0  1594849273   
448         448                   qostanay      KZ           0  1594849274   
468         468                    dasoguz      TM           0  1594849276   
483         483                       lata      PT           0  1594849816   
485         485                 santa rosa      PH           0  1594849816   

     Humidities   Lats    Lngs  Max_temps  Wind_speeds  
24           53  37.85   40.67      75.20         4.70  
96           78  38.30   21.78      73.40         5.41  
158          66  37.32   -8.80      70.23         3.40  
181          90  15.70  120.61      77.22         3.15  
201          78  -9.60  -35.96      77.00         6.93  
276          76  60.06   71.48      72.54         7.58  
313          78  38.30   21.78      73.40         5.41  
396          83  17.49  120.43      77.97         4.32  
435          57  40.47   50.33      78.80         9.17  
448          49  53.17   63.58      73.40         4.47  
468          34  41.84   59.97      78.80         9.17  
483          60  40.16   -8.33      72.00         2.24  
485          83  17.49  120.43      77.97         4.32

In [22]:
hotel_df =filtered_weather_df

hotel_df['Hotel Name'] = np.nan
hotel_df
target_search = "Hotel"
target_radius = 5000
target_type = "lodging"

idx = 0
for index, row in hotel_df.iterrows():
    # set up a parameters dictionary
    params = {
        "location": f"{row['Lats']}, {row['Lngs']}",
        "keyword": target_search,
        "radius": target_radius,
        "type": target_type,
        "key": g_key
        
    }

    # base url
    base_url = "https://maps.googleapis.com/maps/api/place/nearbysearch/json"

    # run a request using our params dictionary
    response = requests.get(base_url, params=params)
    response_json = response.json()
    results = response_json['results']
    if len(results) > 0:
        name = response_json['results'][0]['name']
        hotel_df.iloc[idx, -1] = name
    idx = idx + 1

narrowed_city_df = hotel_df  
narrowed_city_df.dropna()


NameError: name 'filtered_weather_df' is not defined

In [27]:

# Assign the marker layer to a variable
locations = narrowed_city_df[["Lats", "Lngs"]]
markers = gmaps.marker_layer(locations)


NameError: name 'narrowed_city_df' is not defined

In [28]:
# Using the template add the hotel marks to the heatmap
info_box_template = """
<dl>
<dt>Hotel Name</dt><dd>{Hotel Name}</dd>
<dt>City</dt><dd>{City name}</dd>
<dt>Country</dt><dd>{Country}</dd>
</dl>
"""

# Store the DataFrame Row
hotel_info = [info_box_template.format(**row) for index, row in narrowed_city_df.iterrows()]

# Create hotel symbol layer
hotel_layer = gmaps.symbol_layer(
    locations, fill_color='rgba(0, 150, 0, 0.4)',
    stroke_color='rgba(0, 0, 150, 0.4)', scale=2,
    info_box_content=hotel_info
)
# Add the layer to the map
fig.add_layer(markers)
fig.add_layer(hotel_layer)
fig

NameError: name 'narrowed_city_df' is not defined